src/components/search/search.service.ts

```diff
@@ -146,12 +147,16 @@ export class SearchService {
       if (!hydrator) {
         return null;
       }
-      const obj = await hydrator(...args);
-      return {
-        ...obj,
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        __typename: type,
-      };
+      try {
+        const obj = await hydrator(...args);
+        return {
+          ...obj,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          __typename: type,
+        };
+      } catch {
+        return null;
+      }
```

Comment:
Swallowing all errors here is not a good idea. What are we really trying to avoid? I assume it's when a query throws a not found error due to permissions. We should check for that explicitly.

Reply:
syntactically how do I check which type of error this is thrown?

Reply:
With an `instanceof` check

Reply:
Thanks, error caught